core/src/main/java/org/apache/calcite/rel/metadata/RelMdUniqueKeys.java

```diff
@@ -226,6 +227,18 @@ private RelMdUniqueKeys() {}
     return ImmutableSet.of();
   }
 
+  public Set<ImmutableBitSet> getUniqueKeys(TableScan rel, RelMetadataQuery mq,
+      boolean ignoreNulls) {
+    final List<ImmutableBitSet> keys = rel.getTable().getKeys();
+    if (keys.isEmpty()) {
+      return null;
+    }
+    for (ImmutableBitSet key : keys) {
+      assert rel.getTable().isKey(key);
+    }
```

Comment:
Do we really need L236~238?

Reply:
In the case custom table overrides both getKeys() and isKey() method, add this to make sure they are consistent.